### ISSUE TITLE
aria-label on div surrounding svg object

### ISSUE THREAD

**Issue by ethancarlsson** [validator/validator#1194]
SVGs can cause a lot of difficulty for screen readers and are often hidden with aria-hidden. To make the content of the svg available to screen readers a common pattern is 
```html
<div aria-label="description of image content">
    <object data="img.svg" type="image/svg+xml" aria-hidden="true">
        <img src="fallback.png" alt="this is a fallback image for the svg object" />
    </object>
</div>
```

Nu HTML checker gives a "Warning: Possible misuse of aria-label" if you use the above HTML

**Comment by scottaohara** [comment]
This pattern is problematic as a named generic element (`div`) has a long history of not being consistently read by screen readers. https://www.tpgi.com/short-note-on-aria-label-aria-labelledby-and-aria-describedby/

ARIA 1.2 prohibits the naming of generic elements like this.  It's an author error to name such elements. The validator is quite correct in calling out this misuse of the attribute.

For the screen reader / browser pairings which would not expose the named generic `div`, the `aria-hidden=true` on the `object` would mean that the fallback `img`, and thus its `alt` would also not be exposed to a screen reader.

If your SVG is not too complex, in that it can be described with concise alternative text, then you could revise this pattern to the following:

```html
<div aria-label="description of image content" role="img"> <!-- 1 -->
    <object data="img.svg" type="image/svg+xml" aria-hidden="true"> <!-- 2 -->
        <img src="fallback.png" alt> <!-- 3 -->
    </object>
</div>
1. addition of role=img, allowing the div to be named and the pattern to be exposed as a graphic.
2. likely no longer need the aria-hidden=true on the object, as role=img should treat 
   its contents as presentational.  might be good to re-verify though to the necessity 
   of the attribute for your use case.
3. alt value can be empty as aria-label describes the image whether it be rendered as svg or img fallback.
```

Regarding troubles with SVGs and screen readers, give this blog post a read - https://tink.uk/accessible-svg-flowcharts/
Ideally if your SVG was complex and should allow for interaction with individual pieces of the SVG, then it would be marked up with the appropriate ARIA attributes to ensure that all users could understand it.  if the SVG's contents were marked up correctly in its source file, then you likely could remove the wrapping `div` all together (as the necessary accessible information would be part of the rendered SVG file).  Then the `img` could have a descriptive `alt` for the fallback.

I'm curious as to what scenario that necessitates this pattern, with the wide support for rendering SVGs these days.

**Comment by ethancarlsson** [comment]
Hi, thank you very much for this explanation. I can't find where I learnt the pattern in the first place, but it may have been from a time before the rendering SVGs was properly supported (or just wasn't a good idea in the first place).
